core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala

```diff
@@ -474,22 +519,171 @@ object ConsumerGroupCommand extends Logging {
       new KafkaConsumer(properties)
     }
 
+    override def resetOffsets(): Map[TopicPartition, OffsetAndMetadata] = {
+      val groupId = opts.options.valueOf(opts.groupOpt)
+      val consumerGroupSummary = adminClient.describeConsumerGroup(groupId, opts.options.valueOf(opts.timeoutMsOpt))
+      consumerGroupSummary.state match {
+        case "Empty" =>
+          val partitionsToReset = getPartitionsToReset(groupId)
+          val preparedOffsets = prepareOffsetsToReset(groupId, partitionsToReset)
+          val execute = opts.options.has(opts.executeOpt)
+          if (execute)
+            getConsumer().commitSync(preparedOffsets.asJava)
+          preparedOffsets
+        case currentState =>
+          printError(s"Assignments can only be reset if the group '$groupId' is inactive, but the current state is $currentState.")
+          Map.empty
+      }
+    }
+
+    private def parseTopicPartitionsToReset(topicArgs: Seq[String]): Iterable[TopicPartition] = topicArgs.flatMap {
+      case topicArg if topicArg.contains(":") =>
+        val topicAndPartitions = topicArg.split(":")
+        val topic = topicAndPartitions(0)
+        topicAndPartitions(1).split(",").map(partition => new TopicPartition(topic, partition.toInt))
+      case topic => getConsumer().partitionsFor(topic).asScala
+        .map(partitionInfo => new TopicPartition(topic, partitionInfo.partition))
+    }
+
+    private def getPartitionsToReset(groupId: String): Iterable[TopicPartition] = {
+      if (opts.options.has(opts.allTopicsOpt)) {
+        adminClient.listGroupOffsets(groupId).keys
+      } else if (opts.options.has(opts.topicOpt)) {
+        val topics = opts.options.valuesOf(opts.topicOpt).asScala
+        parseTopicPartitionsToReset(topics)
+      } else {
+        CommandLineUtils.printUsageAndDie(opts.parser, "One of the reset scopes should be defined: --all-topics, --topic.")
+      }
+    }
+
+    private def parseResetPlan(resetPlanCsv: String): Map[TopicPartition, OffsetAndMetadata] = {
+      resetPlanCsv.split("\n")
+        .map { line =>
+          val Array(topic, partition, offset) = line.split(",").map(_.trim)
+          val topicPartition = new TopicPartition(topic, partition.asInstanceOf[Int])
+          val offsetAndMetadata = new OffsetAndMetadata(offset.asInstanceOf[Long])
+          (topicPartition, offsetAndMetadata)
+        }.toMap
+    }
+
+    private def prepareOffsetsToReset(groupId: String, partitionsToReset: Iterable[TopicPartition]): Map[TopicPartition, OffsetAndMetadata] = {
+      if (opts.options.has(opts.resetToOffsetOpt)) {
```

Comment:
We probably need to check that the provided offset is in range as we do for the shift option. Same for the CSV case.